### ISSUE TITLE
False positive cmp_owned

### ISSUE THREAD

**Issue by dwijnand** [rust-lang-nursery/rust-clippy#2925]
Given

```rust
"abc".chars().filter(|c| c.to_owned() != '"').collect::<String>()
```

Clippy warns:

```rust
 --> src/lib.rs:2:30
  |
2 |     "abc".chars().filter(|c| c.to_owned() != '"').collect::<String>()
  |                              ^^^^^^^^^^^^ help: try: `c`
  |
  = note: #[warn(cmp_owned)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#cmp_owned
```

However that results in:

```rust
error[E0277]: can't compare `&char` with `char`                          ] 0/1: t
 --> src/lib.rs:2:32
  |
2 |     "abc".chars().filter(|c| c != '"').collect::<String>()
  |                                ^^ no implementation for `&char == char`
  |
  = help: the trait `std::cmp::PartialEq<char>` is not implemented for `&char`
```

**Comment by JoshMcguigan** [comment]
I think this would happen with more than just the char type. This playground link below shows it happening with u8 values. 

https://play.rust-lang.org/?gist=e8860d811b7c69d20a52b12ff5d3cf1f&version=stable&mode=debug&edition=2015

That said, what would be the best way to resolve this?

**Comment by dwijnand** [comment]
Perhaps it should suggest `c != &'"'`?

**Comment by JoshMcguigan** [comment]
I implemented this to suggest `*c`, as that is explicitly what the existing logic was checking for (and then ignoring). I believe a deref is just as good as a borrow on the other side of the equals sign here right?

**Comment by dwijnand** [comment]
I (probably) know less than you so, as far as I'm concerned, if it suggests something that doesn't break compilation, it's a fix. 🙂
